### PR TITLE
Remove `completion` command

### PR DESCRIPTION
**Pull request by ndeloof**
**- What I did**

removed `completion` command
replaced with `--completion` flag (hidden)

**- How I did it**

define a hidden flag, visit flags to check it was set on command line (not just default value)

**- How to verify it**

run
`docker app --completion bash`
`docker app --completion bad`
`docker app --completion`
`docker app`

**- Description for the changelog**
Removed  `completion` command for consistency with docker CLI

**- A picture of a cute animal (not mandatory but encouraged)**
image

### CODE REVIEW

File: internal/commands/root.go [hunk 1]
```diff
@@ -10,15 +11,36 @@ import (
...
+				return nil
+			default:
+				return fmt.Errorf("%q is not a supported shell", completion)
+			}
+		},
 	}
 	addCommands(cmd, dockerCli)
+
+	cmd.Flags().StringVar(&completion, "completion", "", "Generates completion scripts for the specified shell (bash or zsh)")
+	_ = cmd.Flags().MarkHidden("completion")
```
Comment (silvin-lubecki) [inline internal/commands/root.go]: I think this error should be checked 🤔

Comment (ndeloof) [reply internal/commands/root.go]: MarkHidden return an error is flag is not found, which is crazy as we just built it but StringVar doens't return the produced flag

Comment (silvin-lubecki) [reply internal/commands/root.go]: I know, the API is not great on that point, but I think we should still stick to the go mantra: check every error 😅

File: internal/commands/root.go [hunk 2]
```diff
@@ -10,15 +12,40 @@ import (
...
+				return fmt.Errorf("%q is not a supported shell", completion)
+			}
+		},
 	}
 	addCommands(cmd, dockerCli)
+
+	cmd.Flags().StringVar(&completion, "completion", "", "Generates completion scripts for the specified shell (bash or zsh)")
+	err := cmd.Flags().MarkHidden("completion")
+	if err != nil {
+		_, _ = fmt.Fprintf(os.Stderr, "Failed to register command line options: %v", err.Error())
```
Comment (silvin-lubecki) [inline internal/commands/root.go]: Can't we just return a command and an error? And let the caller print the actual message?

Comment (ndeloof) [reply internal/commands/root.go]: We can't, as the caller is `Plugin.Run`  which deine a `func(command.Cli) *cobra.Command` and does not expect errors to happen bulding commands

Comment (silvin-lubecki) [reply internal/commands/root.go]: Oh right... That's a pity... Ok then 👍

File: internal/commands/root.go [hunk 3]
```diff
@@ -10,15 +12,40 @@ import (
...
+			default:
+				return fmt.Errorf("%q is not a supported shell", completion)
+			}
+		},
 	}
 	addCommands(cmd, dockerCli)
+
+	cmd.Flags().StringVar(&completion, "completion", "", "Generates completion scripts for the specified shell (bash or zsh)")
+	err := cmd.Flags().MarkHidden("completion")
+	if err != nil {
```
Comment (silvin-lubecki) [inline internal/commands/root.go]: Just a nit picking then:
```go
if err := cmd.Flags().MarkHidden("completion"); err != nil{
...
}
```

**Pull request merged**